### PR TITLE
[RW-11176][risk=no] Add consumer test between AoU and Leo

### PR DESCRIPTION
**Pull request by evrii**
This PR serves as a means of AoU Workbench to write a contract test against Leonardo.

AoU has several dependencies on Leo. This test serves as a means of verifying a single interaction with Leo, namely the use of the getApp endpoint ({Leo_Base_URL}/api/google/v1/apps/{Name_Of_Google_Project}/{Name_Of_App}). 

This contract test, ensures that AoU's understanding of the endpoint (getApp) that it is calling (consuming) is the same as Leo (the team that is providing the endpoint).

This test verifies (AoU's expectations):
- The endpoint exists
- The response body has the correct attributes and types
- The status code matches expectations
- The response headers match expected values

This test does not verify:
- The endpoint responds with the correct data (in this case an app).

The tests expectations are formalized in something called a contract (api/build/pacts/aou-leonardo.json) that is generated when the test ([AppsApiTest.java](https://github.com/all-of-us/workbench/blob/7b55c989c8ccce57a6a5a5b1b54af01aa2ce3119/api/src/test/java/org/pmiops/workbench/consumer/AppsApiTest.java)) is run. In order to test this PR, please run this test and ensure that contract is generated.

This contract is not stored in this repository. The contract is currently stored in the Leonardo repository, because [they will use this contract to verify AoU's expectations against their system](https://github.com/DataBiosphere/leonardo/pull/3972). However, this is an interim solution. Ideally, contracts will be stored in something called a [pact broker](https://broadworkbench.atlassian.net/wiki/spaces/IRT/pages/2681143308/Pact+Broker+Infrastructure). The consumer (in this case AoU) will send an updated contract to the pact broker every time it updates the corresponding test (in this case [AppsApiTest.java](https://github.com/all-of-us/workbench/blob/7b55c989c8ccce57a6a5a5b1b54af01aa2ce3119/api/src/test/java/org/pmiops/workbench/consumer/AppsApiTest.java)) This will ensure that Leo(the provider) will always have the most up to date version of the contract without relying on manually committing the contract to Leo's repository. Setting up the integration with Pact Broker will done in [RW-11174](https://precisionmedicineinitiative.atlassian.net/browse/RW-11174).

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [x] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

[RW-11174]: https://precisionmedicineinitiative.atlassian.net/browse/RW-11174?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by Peter-Lavigne — APPROVED** [review]
Great job! This test is much simpler than I'd expect. It gives me hope that adding additional contract tests will be low-effort.

I was able to run the test directly from IntelliJ with no extra setup, which was nice.

Out of curiosity, when we start using a broker, what happens the first time we run a contract test? If we created an invalid contract test (for example, calling non-existent endpoints), does that prevent the provider from passing / releasing?

**Review by jmthibault79 — APPROVED** [review]
Pretty nice!

I confirmed that the contract JSON file was created

File: api/src/test/java/org/pmiops/workbench/consumer/AppsApiTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                      body.object(
+                          "cloudContext", context -> context.stringType("cloudprovider", null));
+                    })
+                .build())
+        .toPact();
+  }
+
+  @Test
+  @PactTestFor(pactMethod = "getApp")
+  void getProductById_whenProductWithId10Exists(MockServer mockServer) throws ApiException {
```
[the remark this answers is not in the document]
Comment (evrii) [reply api/src/test/java/org/pmiops/workbench/consumer/AppsApiTest.java:54]: Whoops, I would agree 😅

What do you think of `testGetAppWhenAppExists`?

File: api/src/test/java/org/pmiops/workbench/consumer/AppsApiTest.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
+package org.pmiops.workbench.consumer;
```
[the remark this answers is not in the document]
Comment (evrii) [reply api/src/test/java/org/pmiops/workbench/consumer/AppsApiTest.java:1]: That's a fair point. What do you think of `api/src/test/java/org/pmiops/workbench/pact/consumer/AppsApiTest.java`

I am also partial to `api/src/contracttest/consumer/BigQueryBaseTest.java`
This second option would separate contract tests from unit tests in a way similar to biqquerytest and integration.

What do you think?

[the remark this answers is not in the document]
Comment (jmthibault79) [reply api/src/test/java/org/pmiops/workbench/consumer/AppsApiTest.java]: both are good.  I think I prefer the latter for greater separation.

**Review by jmthibault79 — APPROVED** [review]
./project.rb contracttest works!

**Pull request merged**